### PR TITLE
Security related functionality part three

### PR DESCRIPTION
**Pull request by BolsunovaNataliia**

### CODE REVIEW

File: src/main/java/mate/academy/bookshop/config/SecurityConfig.java [hunk 1]
```diff
@@ -3,17 +3,22 @@
 import static org.springframework.security.config.Customizer.withDefaults;
```
Comment (bhdnchui) [inline src/main/java/mate/academy/bookshop/config/SecurityConfig.java:3]: It's better to place it in `security` package.

File: src/main/java/mate/academy/bookshop/security/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
+            return bearerToken.substring(7);
```
Comment (bhdnchui) [inline src/main/java/mate/academy/bookshop/security/JwtAuthenticationFilter.java]: Let's make constants for strings `Authorization` and  `Bearer ` and get 7 as a length of the last one.

File: src/main/java/mate/academy/bookshop/security/JwtUtil.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    public String generateToken(String username) {
+        return Jwts.builder()
+                .setSubject(username)
+                .setIssuedAt(new Date(System.currentTimeMillis() + expiration))
+                .signWith(secret)
+                .compact();
+    }
+
+    public boolean isValidToken(String token) {
```
Comment (bhdnchui) [inline src/main/java/mate/academy/bookshop/security/JwtUtil.java]: It's better to rename to `isTokenValid`

File: src/main/java/mate/academy/bookshop/service/AuthenticationService.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class AuthenticationService {
+    private final JwtUtil jwtUtil;
+
```
Comment (bhdnchui) [inline src/main/java/mate/academy/bookshop/service/AuthenticationService.java]:
```suggestion
```

File: src/main/java/mate/academy/bookshop/service/AuthenticationService.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
+package mate.academy.bookshop.service;
```
Comment (bhdnchui) [inline src/main/java/mate/academy/bookshop/service/AuthenticationService.java]: It's better to place it in `security` package.

**Review by bhdnchui — CHANGES_REQUESTED** [review]
Good job!
Just a few points to consider.

File: src/main/java/mate/academy/bookshop/service/AuthenticationService.java (hunk 5 above)
Comment (d1sam) [reply src/main/java/mate/academy/bookshop/service/AuthenticationService.java]: still exist

**Review by d1sam — CHANGES_REQUESTED** [review]
Good job, but `src/main/java/mate/academy/bookshop/service/AuthenticationService.java` still need to move `src/main/java/mate/academy/bookshop/security/AuthenticationService.java`

**Review by d1sam — APPROVED** [review]
well done!

**Pull request merged**